First example should be expanded to a complete program

**Example:**
```
// Send a GET request and wait for the response headers.
let mut response = isahc::get("https://example.org")?;
// Read the response body into a string and print it to standard output.
println!("{}", response.text()?);
```

**Error:**
```
error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `std::ops::Try`)
 --> src\http.rs:3:20
  |
1 | / pub fn run() {
2 | | // Send a GET request and wait for the response headers.
3 | | let mut response = isahc::get("https://example.org")?;
  | |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot use the `?` operator in a function that returns `()`
4 | | // Read the response body into a string and print it to standard output.
5 | | println!("{}", response.text()?);
6 | | }
  | |_- this function should return `Result` or `Option` to accept `?`
  |
  = help: the trait `std::ops::Try` is not implemented for `()`
  = note: required by `std::ops::Try::from_error`

error[E0599]: no method named `text` found for struct `http::response::Response<isahc::body::Body>` in the current scope
 --> src\http.rs:5:25
  |
5 | println!("{}", response.text()?);
  |                         ^^^^ method not found in `http::response::Response<isahc::body::Body>`
  |
  = help: items from traits can only be used if the trait is in scope
  = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
          `use isahc::response::ResponseExt;`

error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `std::ops::Try`)        
 --> src\http.rs:5:16
  |
1 | / pub fn run() {
2 | | // Send a GET request and wait for the response headers.
3 | | let mut response = isahc::get("https://example.org")?;
4 | | // Read the response body into a string and print it to standard output.
5 | | println!("{}", response.text()?);
  | |                ^^^^^^^^^^^^^^^^ cannot use the `?` operator in a function that returns `()`
6 | | }
  | |_- this function should return `Result` or `Option` to accept `?`
  |
  = help: the trait `std::ops::Try` is not implemented for `()`
  = note: required by `std::ops::Try::from_error`

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `rust_test1`.
```

**Rust version:**
```
rustc 1.43.1 (8d69840ab 2020-05-04)
```

--- a/README.md
+++ b/README.md
@@ -51,16 +51,28 @@ Safe Rust bindings to libcurl are provided by the [curl crate], which you can us
 
 ## [Documentation]
 
-Please check out the [documentation] for details on what Isahc can do and how to use it. To get you started, here is a really simple example that spits out the response body from https://example.org:
+Please check out the [documentation] for details on what Isahc can do and how to use it. To get you started, here is a really simple, complete example that spits out the response body from https://example.org:
 
 ```rust
-// Send a GET request and wait for the response headers.
-let mut response = isahc::get("https://example.org")?;
-// Read the response body into a string and print it to standard output.
-println!("{}", response.text()?);
+use isahc::prelude::*;
+
+fn main() -> Result<(), isahc::Error> {
+    // Send a GET request and wait for the response headers.
+    // Must be `mut` so we can read the response body.
+    let mut response = isahc::get("http://example.org")?;
+
+    // Print some basic info about the response to standard output.
+    println!("Status: {}", response.status());
+    println!("Headers: {:#?}", response.headers());
+
+    // Read the response body as text into a string and print it.
+    print!("{}", response.text()?);
+
+    Ok(())
+}
 ```
 
-Click [here](https://sagebind.github.io/isahc/isahc/) for built documentation from the latest `master` build.
+Click [here][documentation] for documentation on the latest version. You can also click [here](https://sagebind.github.io/isahc/isahc/) for built documentation from the latest unreleased `master` build.
 
 ## Installation
 

--- a/examples/client.rs
+++ b/examples/client.rs
@@ -1,0 +1,28 @@
+//! Another simple example that creates a custom HTTP client instance and sends
+//! a GET request with it instead of using the default client.
+
+use isahc::{
+    config::RedirectPolicy,
+    prelude::*,
+};
+use std::{
+    io::{copy, stdout},
+    time::Duration,
+};
+
+fn main() -> Result<(), isahc::Error> {
+    // Create a custom client instance and customize a couple things different
+    // than the default settings. Check the documentation of `HttpClient` and
+    // `Configurable` for everything that can be customized.
+    let client = HttpClient::builder()
+        .timeout(Duration::from_secs(5))
+        .redirect_policy(RedirectPolicy::Follow)
+        .build()?;
+
+    let mut response = client.get("https://rust-lang.org")?;
+
+    // Copy the response body directly to stdout.
+    copy(response.body_mut(), &mut stdout())?;
+
+    Ok(())
+}

--- a/examples/http2.rs
+++ b/examples/http2.rs
@@ -1,5 +1,10 @@
-use isahc::config::VersionNegotiation;
-use isahc::prelude::*;
+//! This example simply demonstrates HTTP/2 support by making a request that
+//! enforces usage of HTTP/2.
+
+use isahc::{
+    config::VersionNegotiation,
+    prelude::*,
+};
 
 fn main() -> Result<(), isahc::Error> {
     let response = Request::get("https://nghttp2.org")
@@ -8,7 +13,7 @@ fn main() -> Result<(), isahc::Error> {
         .map_err(Into::into)
         .and_then(isahc::send)?;
 
-    println!("{:?}", response.headers());
+    println!("{:#?}", response.headers());
 
     Ok(())
 }

--- a/examples/json.rs
+++ b/examples/json.rs
@@ -1,5 +1,11 @@
-// This example uses the json feature. As such you should enable it in your Cargo.toml
-// For example, add this line in your `[dependencies]` section : `isahc = { version ="*", features = ["json"]}`
+//! This example uses the `json` crate feature. As such you should enable it in
+//! your Cargo.toml. For example, add this line in your `[dependencies]`
+//! section:
+//!
+//! ```toml
+//! isahc = { version = "*", features = ["json"]}
+//! ```
+
 use isahc::prelude::*;
 
 #[derive(Debug, serde::Serialize)]

--- a/examples/parallel_requests.rs
+++ b/examples/parallel_requests.rs
@@ -2,7 +2,10 @@
 //! simultaneously with no extra cost. Concurrent requests may be made in the
 //! same thread, or from different threads as in this example.
 //!
-//! We're using Rayon here to make parallelism easy.
+//! We're using [rayon] here to make parallelism easy.
+//!
+//! [rayon]: https://github.com/rayon-rs/rayon
+
 use isahc::prelude::*;
 use rayon::prelude::*;
 use std::env;

--- a/examples/progress.rs
+++ b/examples/progress.rs
@@ -1,6 +1,13 @@
 //! A simple command line utility that downloads a file into the void. It
 //! demonstrates how the metrics API can be used to implement an interactive
 //! progress bar.
+//!
+//! Command line options are parsed with [structopt] and the progress bar itself
+//! rendered with [indicatif], both excellent libraries for writing command line
+//! programs!
+//!
+//! [indicatif]: https://github.com/mitsuhiko/indicatif
+//! [structopt]: https://github.com/TeXitoi/structopt
 
 use indicatif::{FormattedDuration, HumanBytes, ProgressBar, ProgressStyle};
 use isahc::prelude::*;

--- a/examples/simple.rs
+++ b/examples/simple.rs
@@ -1,14 +1,18 @@
+//! A very basic example program that sends a GET request and prints out the
+//! response from the server.
+
 use isahc::prelude::*;
 
 fn main() -> Result<(), isahc::Error> {
-    let client = HttpClient::new()?;
-    let mut response = client.get("http://example.org")?;
+    // Send a GET request and wait for the response headers.
+    // Must be `mut` so we can read the response body.
+    let mut response = isahc::get("http://example.org")?;
 
+    // Print some basic info about the response to standard output.
     println!("Status: {}", response.status());
-    println!("Headers:\n{:?}", response.headers());
+    println!("Headers: {:#?}", response.headers());
 
-    // Copy the response body directly to stdout.
-    // std::io::copy(response.body_mut(), &mut std::io::stdout())?;
+    // Read the response body as text into a string and print it.
     print!("{}", response.text()?);
 
     Ok(())

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -65,9 +65,9 @@
 //! # Ok::<(), isahc::Error>(())
 //! ```
 //!
-//! Check out the
+//! For even more examples used in complete programs, please check out the
 //! [examples](https://github.com/sagebind/isahc/tree/master/examples) directory
-//! in the project sources for even more examples.
+//! in the project repo.
 //!
 //! # Feature tour
 //!

